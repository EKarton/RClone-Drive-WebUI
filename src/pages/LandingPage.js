const LandingPage = () => {
  return <div>Welcome!</div>;
};

export default LandingPage;
