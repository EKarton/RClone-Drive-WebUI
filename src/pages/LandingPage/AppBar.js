import MuiAppBar from '@mui/material/AppBar';
import Button from '@mui/material/Button';
import Toolbar from '@mui/material/Toolbar';
import useScrollTrigger from '@mui/material/useScrollTrigger';
import { Link } from 'react-router-dom';
import AppLogo from 'components/AppLogo';
import DarkModeToggleSwitch from 'components/DarkModeToggleSwitch/index';
import './AppBar.scss';

export default function AppBar() {
  const trigger = useScrollTrigger({
    disableHysteresis: true,
    threshold: 0,
  });

  return (
    <>
      <MuiAppBar
        elevation={trigger ? 1 : 0}
        sx={{ bgcolor: 'background.default', color: 'text.primary' }}
      >
        <Toolbar>
          <AppLogo />
          <div className="landing-page-app-bar__right-items">
            <DarkModeToggleSwitch />
            <Button component={Link} to="https://github.com/EKarton/RClone-Drive-WebUI">
              View Github
            </Button>
            <Button variant="contained" component={Link} to="/login">
              Login
            </Button>
          </div>
        </Toolbar>
      </MuiAppBar>
    </>
  );
}
