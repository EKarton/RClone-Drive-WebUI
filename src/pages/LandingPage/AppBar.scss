.landing-page-app-bar {
  &__right-items {
    display: flex;
    column-gap: 1rem;
  }
}
