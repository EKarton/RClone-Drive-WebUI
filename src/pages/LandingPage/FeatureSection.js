import Typography from '@mui/material/Typography';
import cx from 'classnames';
import './FeatureSection.scss';

export default function FeatureSection({ variant, title, subtitle, imgSrc }) {
  const isTextToRight = variant === 'right';
  const isImageToLeft = !isTextToRight;

  return (
    <div
      className={cx('feature-section', {
        'feature-section--right': isTextToRight,
      })}
    >
      <div className="feature-section__text">
        <Typography variant="h4">
          <strong>{title}</strong>
        </Typography>
        <Typography variant="h6">{subtitle}</Typography>
      </div>
      <img
        src={imgSrc}
        alt="Feature"
        className={cx('feature-section__image', {
          'feature-section__image--left': isImageToLeft,
        })}
      />
    </div>
  );
}
