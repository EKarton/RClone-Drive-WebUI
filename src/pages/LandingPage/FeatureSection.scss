.feature-section {
  display: flex;
  flex-direction: row;
  align-items: center;
  max-height: 80vh;
  margin-left: 6rem;
  margin-right: 0;

  &--right {
    flex-direction: row-reverse;
    margin-left: 0;
    margin-right: 6rem;
  }

  &__text {
    width: 35%;
  }

  &__image {
    height: 80vh;
    object-fit: cover;
    overflow: hidden;
    object-position: right;

    &--left {
      object-position: left;
    }
  }
}

@media (max-width: 900px) {
  .feature-section {
    display: block;
    margin: 3rem 0;

    &__text {
      padding: 2rem;
      width: 100%;
      text-align: center;
    }

    &__image {
      width: 100%;
      height: 100%;
    }
  }
}
