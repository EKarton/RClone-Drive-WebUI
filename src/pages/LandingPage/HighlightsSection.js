import InfoIcon from '@mui/icons-material/Info';
import MouseIcon from '@mui/icons-material/Mouse';
import SecurityIcon from '@mui/icons-material/Security';
import Typography from '@mui/material/Typography';
import './HighlightsSection.scss';

export default function HighlightsSection() {
  return (
    <div className="highlights-section">
      <div className="highlights-section__column">
        <div className="highlights-section__header">
          <InfoIcon color="primary" fontSize="large" />
          <Typography variant="h4">
            <strong>About</strong>
          </Typography>
        </div>
        <Typography variant="h6">
          This is a front-facing web application that connects to your RClone instance.
          You can securely view and manage your files across many cloud providers through
          this web app.
        </Typography>
      </div>
      <div className="highlights-section__column">
        <div className="highlights-section__header">
          <SecurityIcon color="warning" fontSize="large" />
          <Typography variant="h4">
            <strong>Secure</strong>
          </Typography>
        </div>
        <Typography variant="h6">
          All actions are done locally on your computer. Nothing is done in our backend
          servers.
        </Typography>
      </div>
      <div className="highlights-section__column">
        <div className="highlights-section__header">
          <MouseIcon color="success" fontSize="large" />
          <Typography variant="h4">
            <strong>Easy to use</strong>
          </Typography>
        </div>
        <Typography variant="h6">
          Drag-and-drop to upload files, right-click and select to delete, move, or copy
          files between different cloud providers.
        </Typography>
      </div>
    </div>
  );
}
