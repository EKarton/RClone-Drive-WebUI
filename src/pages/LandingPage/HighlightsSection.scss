.highlights-section {
  display: flex;
  justify-content: space-around;
  padding: 6rem;
  gap: 3rem;

  &__column {
    flex-grow: 1;
    flex-basis: 0;
  }

  &__header {
    display: flex;
    justify-items: center;
    column-gap: 0.5rem;
    justify-content: left;
    padding-bottom: 1rem;
  }
}

@media (max-width: 900px) {
  .highlights-section {
    flex-direction: column;
  }
}
