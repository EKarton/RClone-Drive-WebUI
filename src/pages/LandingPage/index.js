import CssBaseline from '@mui/material/CssBaseline';
import Toolbar from '@mui/material/Toolbar';
import DarkModeImage from 'assets/images/landing-page/dark-mode.png';
import FileUploadDialogImage from 'assets/images/landing-page/file-upload-dialog.png';
import ActionsImage from 'assets/images/landing-page/move-file-modal.png';
import FileViewerImage from 'assets/images/landing-page/pdf-viewer.png';
import PicturesListPageImage from 'assets/images/landing-page/pictures-list-page.png';
import AppBar from './AppBar';
import FeatureSection from './FeatureSection';
import HighlightsSection from './HighlightsSection';
import IntroSection from './IntroSection';

const LandingPage = () => {
  return (
    <>
      <CssBaseline />
      <AppBar />
      <Toolbar />
      <IntroSection />
      <HighlightsSection />
      <FeatureSection
        variant="right"
        title="View all pictures"
        subtitle="View and scroll through all of your pictures"
        imgSrc={PicturesListPageImage}
      />
      <FeatureSection
        variant="left"
        title="Open and download files"
        subtitle="View, open, and download PDFs, images, and text documents"
        imgSrc={FileViewerImage}
      />
      <FeatureSection
        variant="right"
        title="Upload new files and folders"
        subtitle="Drag-and-drop to upload files and folder to different cloud drives"
        imgSrc={FileUploadDialogImage}
      />
      <FeatureSection
        variant="left"
        title="Manage files on different cloud drives"
        subtitle="Click to copy, rename, delete, and move files between different cloud drives"
        imgSrc={ActionsImage}
      />
      <FeatureSection
        variant="right"
        title="Dark mode"
        subtitle="Ease your eyes with dark theme"
        imgSrc={DarkModeImage}
      />
    </>
  );
};

export default LandingPage;
