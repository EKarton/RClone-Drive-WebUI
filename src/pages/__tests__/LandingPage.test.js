import { customRender } from 'test-utils';
import LandingPage from '../LandingPage';

describe('LandingPage', () => {
  it('should match snapshot', () => {
    const { baseElement } = customRender(<LandingPage />);

    expect(baseElement).toMatchSnapshot();
  });
});
