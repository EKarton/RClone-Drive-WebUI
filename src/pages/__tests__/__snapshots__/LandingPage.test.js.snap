// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match snapshot 1`] = `
<body>
  <div>
    <div>
      Welcome!
    </div>
  </div>
</body>
`;
